rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { mount } from "@vue/test-utils";
+import DateTimePicker from "../dateTimePicker.vue";
+import moment from "moment";
+const mountDateTimePicker = async (props = {}) => {
+  return mount(DateTimePicker, {
+    props: {
+      modelValue: moment().format(),
+      dateClass: "date-class",
+      timeClass: "time-class",
+      ...props,
+    },
+    global: {
+      stubs: {
+        "date-picker": true,
+        "time-picker": true,
+      },
+    },
+  });
+};
+describe("DateTimePicker.vue", () => {
+  let now;
+  beforeEach(() => {
+    now = moment();
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it("sets initial data correctly", async () => {
+    const wrapper = await mountDateTimePicker();
+    expect(wrapper.vm.dateString).toBe(now.format("YYYY-MM-DD"));
+    expect(wrapper.vm.time).toBeInstanceOf(Date);
+  });
+  it("emits update:modelValue when time changes", async () => {
+    const wrapper = await mountDateTimePicker();
+    const newTime = new Date();
+    const newTimeFormatted = moment(newTime).format();
+    await wrapper.setData({ time: newTime });
```

Comment:
recommend replacing the wrapper.setData with an emit from timePicker (`update:modelValue`), ie:

timePicker.vm.$emit(`update:modelValue`, newTime)